### PR TITLE
Add chain upgrade instructions for v0.4.2

### PR DESCRIPTION
**Pull request by alijnmerchant21**
## Summary by CodeRabbit

- **Documentation**
	- Introduced an upgrade guide for the Warden protocol (version 0.4.2) to assist users with the transition process.
	- Provided detailed instructions for both automatic and manual upgrade options, including necessary precautions and recovery strategies.

### CODE REVIEW

File: docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+The method for taking a snapshot depends on your infrastructure, but generally, you can achieve this by backing up the **wardend** directory. If you're [using Cosmovisor](#option-1-use-cosmovisor) to upgrade, it'll back up your data by default.
+
+:::warning
+After stopping the `wardend` process, it's critically important to back up the `.wardend/data/priv_validator_state.json` file.
+:::
+
+**Note**: The **priv_validator_state.json** file is updated every block as your validator participates in consensus rounds. In case the upgrade fails, and you have to restart the previous chain, you'll need this file to avoid double signing. See [Handling node restarts](#handling-node-restarts).
+
+## 2. Run Connect sidecar
```
Comment (jlehtimaki) [inline docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md]: Not needed to mention anymore

File: docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+   mkdir -p  $WARDEN_HOME/cosmovisor/upgrades/v042/bin
+   cp $(which wardend) $WARDEN_HOME/cosmovisor/upgrades/v042/bin
+   ```
+
+   **Note**: Alternatively, enable auto-download in Cosmovisor, and it'll automatically download the binary.
+
+2. Start the node:
+
+   ```shell
+   cosmovisor run start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
```
Comment (jlehtimaki) [inline docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md]:
```suggestion
   cosmovisor run start --home $DAEMON_HOME
```

Comment (jlehtimaki) [reply docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md]: Why do we recommend doing this? :thinking:

Comment (alijnmerchant21) [reply docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md]: Yeah - agreed. We don't need to skip state check.

**Review by pitasi — APPROVED** [review]

File: docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+## Overview
+
+This guide provides steps for upgrading to the [Warden v0.4.2](https://github.com/warden-protocol/wardenprotocol/releases/tag/v0.4.2) release, which makes changes to `keychainFees` to be non nullable.
+
+The network upgrade will take place on **September 2 2024**. The Warden team is going to take the snapshot, export the state, and revert it back if needed.
+
+If you have any outstanding questions, [join our Discord](https://discord.com/invite/warden).
+
+:::tip
+Stay tuned for the precise upgrade date and additional details to be clarified.
```
Comment (ijonele) [inline docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md:16]: Let's delete this tip for this a guide AND for the previous version as well

File: docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+sidebar_position: 2
+---
+
+# v0.4.2
+
+## Overview
+
+This guide provides steps for upgrading to the [Warden v0.4.2](https://github.com/warden-protocol/wardenprotocol/releases/tag/v0.4.2) release, which makes changes to `keychainFees` to be non nullable.
+
+The network upgrade will take place on **September 2 2024**. The Warden team is going to take the snapshot, export the state, and revert it back if needed.
```
Comment (ijonele) [inline docs/developer-docs/docs/operate-a-node/buenavista-testnet/upgrade/v0.4.2.md:11]: Could you please add a comma befor the year both in this guide and in the previous version?

August 13, 2024
September 2, 2024

**Pull request merged**